react/components/ButtonPlain/index.tsx

```diff
@@ -0,0 +1,224 @@
+      )
+    }
+  }, [])
+
+  const {
+    size,
+    variation,
+    children,
+    icon,
+    isLoading,
+    href,
+    target,
+    rel,
+    referrerPolicy,
+    download,
+    noWrap,
+  } = props
+
+  const disabled = props.disabled || isLoading
+  const iconOnly = icon || props.iconOnly
+  const horizontalPadding = iconOnly ? 1 : 2
+  const horizontalCompensation = `nr${horizontalPadding} nl${horizontalPadding} `
+
+  const classes = classNames(
+    'vtex-button bw1 ba fw5 v-mid relative pa0 br2 bn',
+    horizontalCompensation,
+    {
+      't-action--small': size === SIZE.SMALL,
+      't-action': size === SIZE.REGULAR,
+      't-action--large': size === SIZE.LARGE,
+    },
+    { 'icon-button dib': iconOnly },
+    {
+      'bg-transparent b--transparent c-disabled': disabled,
+      'bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary':
+        !disabled && variation === VARIATION.PLAIN,
+      'bg-transparent b--transparent c-on-base--inverted':
+        !disabled && variation === VARIATION.INVERTED,
+      'bg-transparent b--transparent c-danger hover-bg-danger--faded hover-b--danger--faded':
+        !disabled && variation === VARIATION.DANGER,
+    },
+    { pointer: !disabled },
+    { 'inline-flex items-center no-underline': href }
+  )
+
+  const labelClasses = classNames(
+    'flex items-center justify-center h-100 pv1 ttn ',
+    `ph${horizontalPadding}`,
+    { nowrap: noWrap }
+  )
+
+  const style: { fontSize?: number } = {}
+
+  if (iconOnly) {
+    style.fontSize = 0
+  }
+
+  const linkModeProps = {
+    target,
+    rel,
+    referrerPolicy,
+    download,
+  }
+
+  const Element = href ? 'a' : 'button'
+
+  return (
+    <Element
+      id={props.id}
+      data-testid={props.testId}
+      autoFocus={iconOnly ? undefined : props.autoFocus}
+      disabled={iconOnly ? undefined : props.disabled}
+      name={iconOnly ? undefined : props.name}
+      value={iconOnly ? undefined : props.value}
+      tabIndex={0}
+      className={classes}
+      href={href}
+      onClick={handleClick}
+      onMouseEnter={props.onMouseEnter}
+      onMouseLeave={props.onMouseLeave}
+      onMouseOver={props.onMouseOver}
+      onMouseOut={props.onMouseOut}
+      onMouseUp={props.onMouseUp}
+      onMouseDown={props.onMouseDown}
+      onFocus={props.onFocus}
+      onBlur={props.onBlur}
+      ref={props.forwardedRef}
+      style={style}
+      // Button-mode exclusive props
+      type={href ? undefined : (props.type as 'button' | 'submit' | 'reset')}
+      // Link-mode exclusive props
+      {...(href && linkModeProps)}>
+      {isLoading ? (
+        <Fragment>
+          <span className="top-0 left-0 w-100 h-100 absolute flex justify-center items-center">
+            <Spinner
+              secondary={variation === 'primary' || variation === 'danger'}
```

Comment:
```suggestion
              secondary={variation === VARIATION.PRIMARY || variation === VARIATION.DANGER}
```

Reply:
And add `PRIMARY` to the constants 🙂

Reply:
Oh, nice catch. There's no variation `primary` for this component, this unintentionally remained when we wrote this code based on the `Button` component.